src/main/java/de/mari_023/fabric/ae2wtlib/mixin/ContainerTypeBuilderMixin.java

```diff
@@ -37,14 +40,18 @@ public void serverPacketData(PlayerEntity player, ContainerLocator locator, Call
         if(it.isEmpty()) return;
 
         String currentTerminal = WUTHandler.getCurrentTerminal(it);//get the current Terminal, we need to differentiate to return a different WxTgUIObject
-        //TODO do something generic, I don't want to hardcode everything
-        if(hostInterface.isAssignableFrom(WCTGuiObject.class) && currentTerminal.equals("crafting"))
-            cir.setReturnValue(hostInterface.cast(new WCTGuiObject((ItemWT) it.getItem(), it, player, locator.getItemIndex())));
-
-        if(hostInterface.isAssignableFrom(WPTGuiObject.class) && currentTerminal.equals("pattern"))
-            cir.setReturnValue(hostInterface.cast(new WPTGuiObject((ItemWT) it.getItem(), it, player, locator.getItemIndex())));
+        I result = terminalCheck(WCTGuiObject.class, currentTerminal, "crafting", it, player, locator);
+        if (result == null)
+            result = terminalCheck(WPTGuiObject.class, currentTerminal, "pattern", it, player, locator);
+        if (result == null)
+            result = terminalCheck(WITGuiObject.class, currentTerminal, "interface", it, player, locator);
+        cir.setReturnValue(result);
+    }
```

Comment:
I like this solution!
there is just one problem: it doesn't work, and I don't know why.
I do however have a solution to this which I think is even better which will push soon

Reply:
also, this would probably have broken every other ae2 screen

Reply:
(will be resolved by [#1](https://github.com/Domaman202/AE2WirelessTerminalLibrary/pull/1))